src/Npgsql/Internal/TypeHandlers/HstoreHandler.cs

```diff
@@ -54,9 +54,9 @@ public int ValidateAndGetLength(IDictionary<string, string?> value, ref NpgsqlLe
             totalLen += 8;   // Key length + value length
             if (kv.Key == null)
                 throw new FormatException("HSTORE doesn't support null keys");
-            totalLen += _textHandler.ValidateAndGetLength(kv.Key, ref lengthCache, null);
+            totalLen += ((INpgsqlTypeHandler<string>)_textHandler).ValidateAndGetLength(kv.Key, ref lengthCache, null);
```

Comment:
And these casts also to reduce code bloat? It's just `_textHandler` should already implement `INpgsqlTypeHandler<string>` so I don't see what's the point...

Reply:
These are indeed redundant